### PR TITLE
[IMP] add eslintrc check via pre-commit and github actions

### PR DESCRIPTION
**Pull request by yelizariev**
Do you hear this sound? It's the future knocking at your door.

How to use pre-commit locally:

```
  # install pre-commit
  pip install pre-commit

  # stage updates
  git add -A

  # run checks
  pre-commit run

  # review changes made by pre-commit
  git diff

  # add necessary changes to commit
  git add ...

```

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by yelizariev** [comment]
Tested here: https://github.com/yelizariev/------these-list-of-forks-are-useless-------I-hope-github-will-stop-showing-them-some-day---------/pull/27/

# Diff
image

# Checks

image

# Details
image

**Comment by yelizariev** [comment]
Here you can find an example why adding just pre-commit config may not work: https://github.com/odoo/upgrade/pull/1849#issuecomment-712735037

**Comment by yelizariev** [comment]
@qsm-odoo :)

**Comment by yelizariev** [comment]
@Yajo would you like to make a review? :)

**Comment by yelizariev** [comment]
[I took slightly more than 1 minute to check PR for lints
](https://github.com/yelizariev/------these-list-of-forks-are-useless-------I-hope-github-will-stop-showing-them-some-day---------/pull/27/checks?check_run_id=1419341195
)
image

**Comment by qsm-odoo** [comment]
Ah ah :) As mush as I'd like to enforce all Odoo users to use my config file, I cannot take that decision by myself :smile: The guidelines in the wiki were decided in the framework team some years ago. I added a config file in there as a *sample* one to respect the mentioned conventions... and some of my own :smile: (but I only report the errors that do not respect the official guidelines in my reviews :wink:). That sample config could really be reviewed to be cleaner too.

Aside from that, I am actually not a fan of pre-commit if it blindly enforces the conventions... I feel like the developer should have full control over what he writes and this is its responsibility that it follows conventions, *helped* by a linter... especially since yet lots of our files are filled with linter errors.

Anyways, not my decision here... I think @ged-odoo @aab-odoo were discussing something last year to enforce JS conventions, I don't know how that went :wink:

**Comment by yelizariev** [comment]
> Aside from that, I am actually not a fan of pre-commit if it blindly enforces the conventions... I feel like the developer should have full control over what he writes and this is its responsibility that it follows conventions, helped by a linter... especially since yet lots of our files are filled with linter errors.

That's why this github action checks only updated files. Also, I assume that this check will be optional, i.e. PR still can be merged with red lint check

As for config, it can be adjusted later when we get examples why something setting should be turn on or turn off

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{
```
Comment (Yajo) [inline .eslintrc.json:1]: I'd rather use yaml here. It's supported also by eslint and is a much richer syntax that allows i.e. comments.

Also you should disable any style-related checks from the file, and check only AST as explained in https://github.com/OCA/maintainer-quality-tools/pull/618#issuecomment-558576890.

You can draw inspiration from https://github.com/Tecnativa/oca-addons-repo-template/blob/99a27b5597a2bbfa6bd52f023556fdb188a4f567/.eslintrc.yml

File: .github/workflows/pr-lints.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  pull_request:
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+        with:
+          python-version: "3.7.x"
```
Comment (Yajo) [inline .github/workflows/pr-lints.yml:13]: Modernize today; it'll stay frozen for years:

```suggestion
          python-version: "3.9.x"
```

File: .pre-commit-config.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+repos:
+  - repo: https://github.com/pre-commit/mirrors-eslint
```
Comment (Yajo) [inline .pre-commit-config.yaml:2]: And once you've disabled all style checks for eslint, which is unhelpful, leave style to be handled by prettier, which does a better job in that regard:

```suggestion
  - repo: https://github.com/pre-commit/mirrors-prettier
    rev: v2.1.2
    hooks:
      - id: prettier
  - repo: https://github.com/pre-commit/mirrors-eslint
```

Some important points:

1. I don't think Odoo would like this for v12.
1. Different pre-commit pinned versions across odoo branches would difficult the bw/fw ports.
1. It'd be great if Odoo includes a few more checks or formatters. For python there's https://github.com/akaihola/darker if they don't want to fully black the code.
1. Prettier has a XML plugin we use at OCA that works pretty well too. See https://github.com/OCA/oca-addons-repo-template/pull/41/files#diff-0f76a5aa80b9c2ce9f078384ac3e4bccae3defa883c7183632932e624765de51R70-R79

Comment (yelizariev) [reply .pre-commit-config.yaml:2]: 1. A solution could be marking check as failed only when lines changed by pre-commit are intersect with lines changed in pull request.

2. Good point. It means that on updating pre-commit config it has to be done in all maintained branches

**Comment by C3POdoo** [comment]
Dear @yelizariev,

Thank you for your contribution but the version 12.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._

**Pull request closed without merging**